### PR TITLE
Publicize: Configure Publicize only if the Publicize Module is active. 

### PR DESCRIPTION
**Pull request by spsiddarthan**
We are loading in Publicize in Jetpack, even if the Publicize module is inactive. We need to load Publicize only if the module is active. 

To fix p1654201798873139-slack-CDD9LQRSN and https://github.com/Automattic/jetpack/pull/23871

#### Changes proposed in this Pull Request:

* Configure Publicize in Jetpack only if the Publicize Module is active. 

#### Jetpack product discussion

https://github.com/Automattic/jetpack/pull/23871#issuecomment-1140746502

#### Testing instructions:

- Install Jetpack and make sure Publicizing works correctly
- Deactivate the Publicize module and make sure all the Publicize elements get hidden on the block editor.
- Activate the Publicize module again and make sure you're able to Publicize correctly.
- Deactivate the Publicize module and deactivate Jetpack, activate the Jetpack Social Plugin and make sure you're able to - Publicize via the Classic Editor.

### CODE REVIEW

File: projects/plugins/jetpack/tests/php/core-api/wpcom-fields/test-post-fields-publicize-connections.php [hunk 1]
```diff
@@ -83,6 +83,10 @@ public static function wpSetUpBeforeClass( $factory ) {
 		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
 			self::setup_connections_wpcom();
 		} else {
+			global $publicize_ui;
```
Comment (pablinos) [inline projects/plugins/jetpack/tests/php/core-api/wpcom-fields/test-post-fields-publicize-connections.php:86]: Should we also unset the global in these tests?

**Review by pablinos — DISMISSED** [review]
This works for me. I can see that the classes aren't instantiated with the module disabled, and the post field isn't being processed with a REST request for the post.

**Review by jeherve — APPROVED** [review]
This looks good to me. 🚢

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-11.0` in 64ba182f671a03e1d1c08a6fe9ca795efbd3a576

**Comment by spsiddarthan** [comment]
r246833-wpcom

**Pull request merged**